docs/specifications/tests/DNSSEC-TP/dnssec01.md

```diff
@@ -1,41 +1,120 @@
-## DNSSEC01: Legal values for the DS hash digest algorithm
+# DNSSEC01: Legal values for the DS hash digest algorithm
 
-### Test case identifier
-**DNSSEC01** Legal values for the DS hash digest algorithm
+## Test case identifier
+**DNSSEC01**
 
-### Objective
+## Objective
 
-The allowed Digest Algorithms in a DS record published by the parent are
-published by IANA in [Delegation Signer (DS) Resource Record (RR) Type
-Digest Algorithms](https://www.iana.org/assignments/ds-rr-types/ds-rr-types.xml).
-No other DS Digest Algorithm values other than those allocated by IANA
-should be published in DNS.
+The list of allowed Digest Algorithms in a DS record published by 
+the parent is specified by [RFC 8624], section 3.3, and is published 
+in the [IANA registry] of *DS RR Type Digest Algoritms*. No DS 
+Digest Algorithm values, other than those specified in the RFC and
+allocated by IANA, should be used in public DNS.
 
-The use of the DS RR type is described in section 2.4 of
-[RFC 4035](https://tools.ietf.org/html/rfc4035#section-2.4), and the exact
-definition of the DS RR type is in
-[RFC3658](https://tools.ietf.org/html/rfc3658).
+If [RFC 8624] and the [IANA registry] disagree on the same DS digest 
+algorithm, the RFC takes precedence until the registry has a been 
+updated with a reference to the RFC.
 
-### Inputs
+## Inputs
 
-The domain name to be tested.
+* "Child Zone" - The domain name to be tested.
+* "Algorithm Status" - The status of all DS digest algorithms from 
+  [RFC 8624] and the [IANA registry].
+* "Test Type" - The test type with value "undelegated" or "normal".
 
-### Ordered description of steps to be taken to execute the test case
+* "Undelegated DS" - The DS record or records submitted
+  (only if *Test Type* is undelegated).  
 
-1. Retrieve the DS RR set from the parent zone.
-2. From the DS RRs, extract the Digest type values.
-3. If any value from the Digest type is not included in the IANA table
-   (currently values 1 to 4 are valid), this test case fails.
+## Ordered description of steps to be taken to execute the test case
 
-### Outcome(s)
+1. If the *Test Type* is "undelegated, then:
```

Comment:
If we are going this path, then shouldn't we add an undelegated section for all test cases?

Reply:
Yes, in many cases, or maybe most cases. When we start using the [new methods] it can often be hidden from the explicit test case text.

See updated [Basic01] (in develop branch) which was approved 5 days ago in PR #642, which heavily refers to undelegated test.

I see no other way to handle it.

[new methods]: https://github.com/zonemaster/zonemaster/blob/develop/docs/specifications/tests/MethodsNT.md

[Basic01]: https://github.com/zonemaster/zonemaster/blob/develop/docs/specifications/tests/Basic-TP/basic01.md